lib/measured/unit_system.rb

```diff
@@ -1,55 +1,67 @@
 class Measured::UnitSystem
-  attr_reader :units
+  class << self
+    def units
+      raise "`UnitSystem` does not have a `units` object. You cannot directly subclass `UnitSystem`. Instead, build a new unit system by calling `Measured.build`."
+    end
 
-  def initialize(units)
-    @units = units.map { |unit| unit.with_unit_system(self) }
-  end
+    def new(value, unit)
+      Measured::Measurable.new(value, unit_from_unit_or_name!(unit))
+    end
```

Comment:
So the one weird thing now is that `new` here isn't returning an instance of `UnitSystem`, but rather `Measurable`. This will dramatically simplify migrating to 2.0, but may come as a surprise to someone who digs a little deeper.

I'm not opposed to returning a `Bar` from `Foo#new`, but don't know the expectations or idioms in Ruby well enough to know if this is 🚫 .